src/deeprank_gnn/predict.py

```diff
@@ -139,10 +144,51 @@ def pdb_to_fasta(pdb_file_path: Path, main_fasta_fh: TextIOWrapper) -> None:
         main_fasta_fh.write(f">{root}.{chain.id}\n{sequence}\n")
 
 
+def calculate_checksum(file_path, algo="sha256") -> str:
+    h = hashlib.new(algo)
+    with open(file_path, "rb") as f:
+        for chunk in iter(lambda: f.read(8192), b""):
+            h.update(chunk)
+    return h.hexdigest()
+
+
+def download_weights(url: str, dest: str) -> None:
+    response = requests.get(url, stream=True)
+    response.raise_for_status()
+    with open(dest, "wb") as f:
+        for chunk in response.iter_content(chunk_size=8192):
+            if chunk:
+                f.write(chunk)
+
+
+def fetch_weights() -> str:
+    """Fetch the ESM model weights and regression weights."""
+    models = {
+        "model": {
+            "env": "WEIGHT_PATH",
+            "url": f"https://dl.fbaipublicfiles.com/fair-esm/models/{ESM_MODEL}.pt",
+            "default_filename": f"{ESM_MODEL}.pt",
+        },
+        "regression": {
+            "env": "REG_WEIGHT_PATH",
+            "url": f"https://dl.fbaipublicfiles.com/fair-esm/regression/{ESM_MODEL}-contact-regression.pt",
+            "default_filename": f"{ESM_MODEL}-contact-regression.pt",
+        },
+    }
+
+    for _, info in models.items():
+        path = os.getenv(info["env"])
+        if (
+            not os.path.exists(path)
+            or calculate_checksum(path) != EXPECTED_CHECKSUM[info["default_filename"]]
```

Comment:
Above `os.getenv` might return `None` (https://docs.python.org/3/library/os.html#os.getenv) and your `calculate_checksum` expects `string`. So if the system variable is not defined (which is different to being defined as empty) this ill raise a `TypeError`.  

Quick fix for your code is to override the default:

```python
 path = os.getenv(info["env"], default="")
```